### ISSUE TITLE
Add support for I2C Adafruit servo board?

### ISSUE THREAD

**Issue by samuk** [mgonzs13/ros2_rover#5]
Would you consider adding support for this [16ch servo board ](https://learn.adafruit.com/16-channel-pwm-servo-driver?view=all) & it's [clones](https://www.aliexpress.com/item/33047273435.html)

Here are the [Pi instructions](https://learn.adafruit.com/adafruit-16-channel-servo-driver-with-raspberry-pi/configuring-your-pi-for-i2c)

I'm interested in trying these [waterproof servos](https://www.aliexpress.com/item/1005002297372748.html) but they are not the bus type.

**Comment by mgonzs13** [comment]
Yes, again, it is interesting. In fact, I want to build a new rover. The main problem is that I'll need the hardware components.

**Comment by samuk** [comment]
Since that comment, I've found [FEICHO](https://www.aliexpress.com/item/1005001325059005.html) waterproof servos's which are more affordable at $29.27 US and provide 40kg/cm

I appreciate it would still be a $296 upgrade, for the full Sawppy. It would be worth it to me personally for the waterproof rating.

Would it be possible to do some testing with just the servo driver and one of the servos as proof of concept? I'd be willing to fund a servo and board and get it posted to you.

**Comment by samuk** [comment]
Just spotted that this ROS2 code exists for that servo board: https://github.com/kerry-t-johnson/i2c_pwm

**Comment by mgonzs13** [comment]
I think the Adafruit Python library is easier to use for a first approach.

**Comment by samuk** [comment]
I've just seen this rather nice small 6x6 with servo https://www.youtube.com/@wildwillyrobots

Two of these open hardware drivers https://wiki.seeedstudio.com/Motor_Bridge_Cape_v1.0/ stacked together could drive 4x steering servo, plus six DC drive motors.

At ~£20 each they seem fairly affordable https://uk.rs-online.com/web/p/power-motor-robotics-development-tools/1743284

This is the library for it https://github.com/Seeed-Studio/MotorBridgeCapeforBBG_BBB

**Comment by mgonzs13** [comment]
Hi @samuk, thanks for the info. We are working on a new rover and we revisiting the motors focusing mainly on steppers. Do you have any specific motor in mind?

**Comment by samuk** [comment]
No, but this project: https://www.thingiverse.com/thing:5363032 Uses these https://www.aliexpress.com/item/1005002662047047.html

BLDC would be nice too of course! Perhaps using https://github.com/Twisted-Fields/rp2040-motor-controller drivers  https://github.com/runger1101001/rp2040-motor-controller-firmware

**Comment by samuk** [comment]
Just to add to the I2C idea I've recently been playing with https://www.smart-prototyping.com/Zio-4-DC-Motor-Controller.html

2x of them could be used to handle the 6x power motors, with the steering done by servo.
